lib/middleware/resource.js

```diff
@@ -4,12 +4,16 @@ function factory ({ loader }) {
   return async (req, res, next) => {
     let resources = await loader.forClassOperation(req.hydra.term)
 
+    if (resources.length > 1) {
+      return next(new Error(`no unique resource found for: <${req.hydra.term.value}>`))
+    }
+
     if (resources.length === 0) {
       resources = await loader.forPropertyOperation(req.hydra.term)
-    }
+      res.locals.hydra.resourceCandidates = resources
 
-    if (resources.length > 1) {
-      return next(new Error(`no unique resource found for: <${req.hydra.term.value}>`))
+      debug('Multiple resource candidates found')
+      return next()
```

Comment:
Does not error immediately but instead set the potential "property operation" candidates to be filtered by the `operation` middleware